src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs

```diff
@@ -170,12 +170,13 @@ public IWebHostBuilder UseStartup([DynamicallyAccessedMembers(StartupLinkerOptio
         UseSetting(WebHostDefaults.ApplicationKey, startupAssemblyName);
 
         // UseStartup can be called multiple times. Only run the last one.
-        _startupObject = startupType;
+        _builder.Properties[_startupConfigName] = startupType;
```

Comment:
Can you add a test for this change? I believe it can be trigger by having multiple ConfigureWebHost calls with UseStartup inside them.